docs/blog/posts/introduction-to-bpfd.md

```diff
@@ -0,0 +1,605 @@
+
+<!-- more -->
+
+eBPF is a simple and efficient way to dynamically load programs into the kernel
+at runtime, with safety and performance provided by the kernel itself using a
+Just-In-Time (JIT) compiler and verification process. There are a wide variety
+of types of programs one can create with eBPF, which include everything from
+networking applications to security systems.
+
+We are however still fairly early in the eBPF journey and it's not all kittens
+and rainbows: the process of developing, testing, deploying, and maintaining
+eBPF programs is not a road well traveled yet, and the story gets even more
+complicated when you want to deploy your programs into more complicated setups,
+such as a [Kubernetes] cluster. It was these kinds of problems that
+motivated the creation of [bpfd]: a system daemon for loading and managing eBPF
+programs in both traditional systems and Kubernetes clusters. In this blog post,
+we'll discuss the problems bpfd can help solve, and how to deploy and use it.
+
+[userspace]:https://en.wikipedia.org/wiki/User_space_and_kernel_space
+[kmod]:https://wiki.archlinux.org/title/Kernel_module
+[eBPF]:https://ebpf.io
+[linux]:https://kernel.org
+[Kubernetes]:https://kubernetes.io
+[bpfd]:https://bpfd.dev
+
+## Challenges with developing and deploying eBPF programs
+
+eBPF is a revolutionary kernel technology, but it's also still fairly early in
+terms of ecosystem maturity. While some organizations have invested deeply and
+have had success developing, deploying, and maintaining production software
+which includes eBPF programs the barrier to entry is still very high. When it
+comes to developing, testing is difficult, and trying to work through problems
+with your programs reported by the [eBPF verifier] can be a very slow process at
+best, and a hard impediment for progress at worst due to its recondite nature.
+
+Beyond the fundamentals of developing and loading programs locally to test them,
+once you have working code the process of deploying these in testing and
+staging environments to validate them often ends up resulting in a lot of
+custom program loading and management functionality specific to the application.
+Not to mention eventually moving to production systems, potentially in
+environments like Kubernetes clusters: the operational considerations continue to
+compound.
+
+Security is another significant challenge, which we will cover in more depth in
+a follow-on blog.  However, at a high level, applications that use eBPF
+typically load their own eBPF programs, which requires at least CAP_BPF.  Many
+BPF programs and attach points require additional capabilities from
+CAP_SYS_PTRACE, CAP_NET_ADMIN and even including CAP_SYS_ADMIN.  These
+privileges include capabilities that aren’t strictly necessary for eBPF and are
+too coarsely grained to be useful.  Since the pods that load eBPF are mostly
+long-lived and often don’t drop privileges it leaves a wide attack surface.
```

Comment:
So let's be careful about the "context" we're talking about bpf in here, because doesn't the same thing apply to a bpf loading process on a single node as well? 

"Since the pods...."

Makes me feel like we jumped from talking about the challenges of eBPF in general to talking about the challenges of eBPF in Kubernetes.  Having a clear and explicit separation between the two will help with this. 

In this specific case I think we can just say "Since the processes which load ebpf are usually long-lived and often...."